tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/s3/TieredStorageS3Client.java

```diff
@@ -0,0 +1,325 @@
+            }
+        });
+        return completableFuture;
+    }
+
+    public CompletableFuture<Boolean> exist(String prefix) {
+        CompletableFuture<ListObjectsV2Response> listFuture = this.client.listObjectsV2(builder -> builder.bucket(this.bucket).prefix(prefix));
+        return listFuture.thenApply(resp -> {
+            return resp.contents().size() > 0;
+        });
+    }
+
+    public CompletableFuture<Boolean> deleteObject(String key) {
+        CompletableFuture<Boolean> future = new CompletableFuture<>();
+        this.client.deleteObject(builder -> builder.bucket(this.bucket).key(key)).whenComplete((deleteObjectResponse, throwable) -> {
+            if (throwable != null) {
+                LOGGER.error("Delete object from S3 failed, key: {}, region: {}, bucket: {}", key, this.region, this.bucket, throwable);
+                future.complete(false);
+            } else {
+                LOGGER.info("Delete object from S3, key: {}, region: {}, bucket: {}", key, this.region, this.bucket);
+                future.complete(true);
+            }
+        });
+        return future;
+    }
+
+    public CompletableFuture<List<String/*undeleted keys*/>> deleteObjets(final List<String> keys) {
+        if (keys == null || keys.isEmpty()) {
+            return CompletableFuture.completedFuture(Collections.emptyList());
+        }
+        List<ObjectIdentifier> objects = keys.stream().map(key -> ObjectIdentifier.builder().key(key).build()).collect(Collectors.toList());
+        Delete delete = Delete.builder().objects(objects).build();
+        DeleteObjectsRequest deleteObjectsRequest = DeleteObjectsRequest.builder().bucket(this.bucket).delete(delete).build();
+        return this.client.deleteObjects(deleteObjectsRequest).thenApply(resp -> {
+            List<String> undeletedKeys = null;
+            if (resp.deleted().size() != keys.size()) {
+                List<String> deleted = resp.deleted().stream().map(deletedObject -> deletedObject.key()).collect(Collectors.toList());
+                undeletedKeys = keys.stream().filter(key -> !deleted.contains(key)).collect(Collectors.toList());
+            } else {
+                undeletedKeys = Collections.emptyList();
+            }
+            return undeletedKeys;
+        }).exceptionally(throwable -> {
+            LOGGER.error("Delete objects from S3 failed, keys: {}, region: {}, bucket: {}", keys, this.region, this.bucket, throwable);
+            return keys;
+        });
+    }
+
+    public CompletableFuture<List<String>> deleteObjects(String prefix) {
+        CompletableFuture<List<String>> readObjectsByPrefix = this.client.listObjectsV2(builder -> builder.bucket(this.bucket).prefix(prefix)).thenApply(resp -> {
+            return resp.contents().stream().map(s3Object -> s3Object.key()).collect(Collectors.toList());
+        });
+        return readObjectsByPrefix.thenCompose(keys -> {
+            return this.deleteObjets(keys);
```

Comment:
Many lambda functions in this file could be simplified, please follow the suggestions from idea.